src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.ReinferredVariableTracker.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class NullableWalker
+    {
+#nullable enable
+        internal sealed class ReinferredVariableTracker
+        {
+            internal readonly struct Readonly
+            {
+                private readonly ReinferredVariableTracker? _tracker;
+                internal Readonly(ReinferredVariableTracker tracker)
+                {
+                    _tracker = tracker;
+                }
+
+                internal readonly T? GetReinferredSymbol<T>(T originalSymbol) where T : Symbol => _tracker?.GetReinferredSymbol(originalSymbol);
+                internal readonly ReinferredVariableTracker EnterNewFunction()
+                {
+                    Debug.Assert(_tracker is object);
+                    return _tracker?.EnterNewFunction()!;
+                }
+            }
+
+            private readonly Dictionary<Symbol, Symbol> _reinferredVariables;
+            private readonly ReinferredVariableTracker.Readonly _parent;
+
+            private ReinferredVariableTracker(ReinferredVariableTracker.Readonly parent)
+            {
+                _reinferredVariables = new Dictionary<Symbol, Symbol>();
```

Comment:
>new Dictionary<Symbol, Symbol>(); [](start = 39, length = 33)

What comparer would be appropriate for this dictionary? I assume ReferenceEqualityComparer is what we want.

Reply:
Reference is what we want, so I didn't set one. I can add a comment.

---
In reply to: [327852656](https://github.com/dotnet/roslyn/pull/38769#discussion_r327852656) [](ancestors = 327852656)

Reply:
> Reference is what we want, so I didn't set one. I can add a comment.

Shouldn't we use ```ReferenceEqualityComparer``` then? Default comparer is not reference comparer.

---
In reply to: [327852930](https://github.com/dotnet/roslyn/pull/38769#discussion_r327852930) [](ancestors = 327852930,327852656)